# **Avance 1. Análisis exporatorio de datos**

---

Profa. Dra. Grettel Barceló Alonso 

Prof. Dr. Luis Eduardo Falcón Morales 

Profa. Verónica Sandra Guzmán de Valle 

---


**Estudiantes:**


Marcela Alejandra Rosales Jiménez – A01032022

José Antonio Mendoza Castro – A01794067


29 de septiembre de 2024



## WTS

### **Videos**

Importar librerias

In [28]:
import pathlib

Visualizar cuantos archivos en el directorio `wts_dataset_zip`. Antes de correr el siguiente codigo se verifico que ya se hubieran extraido los 4 archivos zip que estaban dentro del dataset.

In [45]:
dataset_path = f"dataset/wts_dataset_zip"

In [46]:
def get_all_files(files_path):
    files_path = pathlib.Path(files_path)
    listed_elements = list(files_path.rglob("*"))                           
    listed_files = [e for e in listed_elements if not e.is_dir()]
    return listed_files

def get_listed_files_with_types(listed_files):
  files_types = {}
  for ff in listed_files:
    ext = ff.suffix
    if not files_types.get(ext):
      files_types[ext] = []
    files_types[ext].append(str(ff))
  return files_types



listed_files = get_all_files(f"{dataset_path}")
print(f"Cantidad de archivos en total = {len(listed_files)}")


files_by_ext = get_listed_files_with_types(listed_files)

for k, v in files_by_ext.items():
    print(f" - Extension {k} tiene {len(v)}")


Cantidad de archivos en total = 14126
 - Extension .csv tiene 1
 - Extension .zip tiene 4
 - Extension  tiene 1
 - Extension .DS_Store tiene 1
 - Extension .json tiene 9907
 - Extension .mp4 tiene 4212


Podemos ver que dentro de el zip del dataset hay 4,212 videos.

Visualmente se evaluan los videos y se decide lo siguiente
- No se utilizarán los videos que estan dentro de `wts_dataset_zip/external` ya que son grabaciones desde la perspectiva del asiento delantero de un carro.
- No se utilizarán los videos que estén dentro de una carpeta llamada `vehicle_view` por el mismo motivo.

In [63]:
files_vehicle_view = []
files_vehicle_view_by_type = {
    'train': [],
    'val': [],
}

for mp4 in files_by_ext['.mp4']:
    if not f"{dataset_path}/external" in mp4 and not "vehicle_view" in  mp4:
        files_vehicle_view.append(mp4)
        if "videos/train" in mp4:
            files_vehicle_view_by_type['train'].append(mp4)
        elif "videos/val" in mp4:
            files_vehicle_view_by_type['val'].append(mp4)
        else:
            print(f"W: No train ni val. Que es? {mp4}")
            

print(f"Cantidad de archivos a utilizar = {len(files_vehicle_view)}")
for k, v in files_vehicle_view_by_type.items():
    print(f" - {k} tiene {len(v)}")


Cantidad de archivos a utilizar = 600
 - train tiene 403
 - val tiene 197


Podemos concluir que para los objetivos de nuestro proyecto tenemos 600 videos mp4, de los cuales el dataset de WTS sugiere usar 403 en videos de entrenamiento y 197 en videos de validación.

In [65]:
files_by_ext['.json']

['dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video1098_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video4764_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video5408_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video1248_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video157_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video6551_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video849_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video7261_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video3453_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video8211_caption.json',
 'dataset/wts_dataset_zip/external/BDD_PC_

### **Anotaciones**

In [66]:
cat 'dataset/wts_dataset_zip/external/BDD_PC_5K/annotations/caption/train/video3453_caption.json'

{
    "id": "93523959-4667-4f32-b979-5bcd4032e19c",
    "video_name": "video3453.mp4",
    "fps": "29.985",
    "event_phase": [
        {
            "labels": [
                "prerecognition"
            ],
            "caption_pedestrian": "The pedestrian is a male in his 20s with a height of 160 cm. He is wearing a yellow T-shirt and gray slacks. It is a weekday and the weather is rainy, making the brightness level dark. The road surface is wet and the road is an urban main road with two-way traffic and sidewalks on both sides. The pedestrian is standing diagonally to the right in front of the vehicle. His body is perpendicular to the vehicle and tilted slightly to the left. He is closely watching his crossing destination and is unaware of the vehicle. The pedestrian is crossing the street, ignoring the signal, at a slightly higher speed. The environment conditions, such as the rain and dark brightness, may affect visibility and road surface conditions, potentially increasing the